### PR TITLE
Feature/26

### LINKED ISSUE #26: [FEATURE] - Anonymous users should be able to track progress of a session
### Reference Issues

#25 

### Summary

As students complete exercises of a session, they should be able to monitor their progress.

As an extension to #25, entailing users validating their solution, the functionality should be extended, such that if a student's solution satisfies all testcases it should be updated in the database.

### Unresolved questions

_No response_

### Definition of Done

When a solution satisfying all testcases of an exercise results in the creation of a `solved` instance in the database, referencing the correct student and exercise.

### PR DESCRIPTION
**Pull request by KristianS93**
## Description

Making sure that a anon user who have joined a session can submit their solution, while also saving this state to the session, such that the frontend can mark an exercise for a user as solved.

### Resolved Issue
Fixes #26

## Changes

Changes made to solutionrunner repo, session repo, services and endpoints. 

## Additional Information

This is a happy path working example, further tests and try outs for different edge cases are still needed.
Behavior of the response and functionality should be discussed by the frontend and backend team. 

For the backend team some general clean up might be necessary and some considerations on how error messages should be propagated from repos.

## Checklist

- [x] New and old tests passed

### CODE REVIEW

File: Core/Solutions/Contracts/ISolutionRunnerService.cs [hunk 1]
```diff
@@ -7,5 +7,6 @@ namespace Core.Solutions.Contracts;
 public interface ISolutionRunnerService
 {
     Task<Result> CreateSolutionAsync(ExerciseSubmissionDto dto);
-    Task<Result> SubmitSolutionAsync(SubmitSolutionDto dto);
+    Task<Result> SubmitSolutionAsync(SubmitSolutionDto dto, int exerciseId, int userId);
```
Comment (JMyrtue) [inline Core/Solutions/Contracts/ISolutionRunnerService.cs:10]: Consider the naming of the above two methods.

Can be confusing as is.

**Review by JMyrtue — APPROVED** [review]
